packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -355,85 +355,115 @@ IOSink? _openSink(String? output) {
   return sink;
 }
 
-/// A generator that will write code to a sink based on the contents of [PigeonOptions].
-abstract class Generator {
-  /// Returns an [IOSink] instance to be written to if the [Generator] should
-  /// generate.  If it returns `null`, the [Generator] will be skipped.
-  IOSink? shouldGenerate(PigeonOptions options);
+/// An adapter that will call a generator to write code to a sink
+/// based on the contents of [PigeonOptions].
+abstract class GeneratorAdapter {
+  /// Constructor for [GeneratorAdapter]
+  GeneratorAdapter(this.fileTypeList);
+
+  /// A list of [FileType]
+  /// Used to tell the generator method which type of file to create.
+  List<FileType> fileTypeList;
+
+  /// Returns an [IOSink] instance to be written to if the [GeneratorAdapter] should
+  /// generate.  If it returns `null`, the [GeneratorAdapter] will be skipped.
+  IOSink? shouldGenerate(PigeonOptions options, FileType fileType);
 
   /// Write the generated code described in [root] to [sink] using the
   /// [options].
-  void generate(StringSink sink, PigeonOptions options, Root root);
+  void generate(
+      StringSink sink, PigeonOptions options, Root root, FileType fileType);
 
-  /// Generates errors that would only be appropriate for this [Generator]. For
-  /// example, maybe a certain feature isn't implemented in a [Generator] yet.
+  /// Generates errors that would only be appropriate for this [GeneratorAdapter]. For
+  /// example, maybe a certain feature isn't implemented in a [GeneratorAdapter] yet.
   List<Error> validate(PigeonOptions options, Root root);
 }
 
 DartOptions _dartOptionsWithCopyrightHeader(
-    DartOptions? dartOptions, String? copyrightHeader) {
-  dartOptions = dartOptions ?? const DartOptions();
+    DartOptions? dartOptions, String? copyrightHeader,
+    [String? dartOutPath, String? testOutPath]) {
```

Comment:
Can we make these named rather than positional? Having positional arguments with the same types makes the callsites harder to understand and easier to mess up.